src/hotspot/share/opto/escape.cpp

```diff
@@ -1811,8 +1828,8 @@ void ConnectionGraph::adjust_scalar_replaceable_state(JavaObjectNode* jobj) {
       PointsToNode* ptn = j.get();
       if (ptn->is_JavaObject() && ptn != jobj) {
         // Mark all objects.
-        jobj->set_scalar_replaceable(false);
-         ptn->set_scalar_replaceable(false);
+        set_not_scalar_replaceable(jobj NOT_PRODUCT(COMMA "is merged with other object"));
+        set_not_scalar_replaceable(ptn NOT_PRODUCT(COMMA "is merged with other object"));
```

Comment:
how about we also dump Java Object(x) here? eg. 
```
JavaObject(8) NoEscape(NoEscape) is NSR. is merged with other Java object(10) 
```

Reply:
That seems like a good idea.